src/OTSubscriber.js

```diff
@@ -57,43 +60,59 @@ export default class OTSubscriber extends Component {
 
     this.subscriberId = uuid();
     const { subscriberId } = this;
-
     const subscriber = this.state.session.subscribe(
-      this.state.stream,
-      container,
-      this.props.properties,
-      (err) => {
-        if (subscriberId !== this.subscriberId) {
-          // Either this subscriber has been recreated or the
-          // component unmounted so don't invoke any callbacks
-          return;
-        }
-        if (err && typeof this.props.onError === 'function') {
-          this.props.onError(err);
-        } else if (!err && typeof this.props.onSubscribe === 'function') {
-          this.props.onSubscribe();
-        }
-      },
-    );
+            this.state.stream,
+            container,
+            this.props.properties,
+            (err) => {
+              if (subscriberId !== this.subscriberId) {
+                    // Either this subscriber has been recreated or the
+                    // component unmounted so don't invoke any callbacks
+                return;
+              }
+              if (err &&
+                this.props.retry &&
+                this.state.currentRetryAttempt < this.maxRetryAttempts) {
+                    // Error during subscribe function
+                this.handleRetrySubscriber();
+                // If there is a retry action, do we want to execute the onError props function?
+                // return;
+              }
+              if (err && typeof this.props.onError === 'function') {
+                this.props.onError(err);
+              } else if (!err && typeof this.props.onSubscribe === 'function') {
+                this.props.onSubscribe();
+              }
+            },
+        );
 
     if (
-      this.props.eventHandlers &&
-      typeof this.props.eventHandlers === 'object'
-    ) {
+            this.props.eventHandlers &&
+            typeof this.props.eventHandlers === 'object'
+        ) {
       subscriber.on(this.props.eventHandlers);
     }
 
     this.setState({ subscriber });
   }
 
+  handleRetrySubscriber() {
+    setTimeout(() => {
+      let { currentRetryAttempt } = this.state;
+      currentRetryAttempt += 1;
+      this.setState({ subscriber: null, currentRetryAttempt });
```

Comment:
Since `setState` get batched, we should not use the state and add one. Should we use the state retrieved from the `setState` callback instead? Something like this:

```
this.setState(state => ({
  currentRetryAttempt: state.currentRetryAttempt + 1
}))
```

Reply:
Good catch